## Importación de las librerías necesarias

In [44]:
import pandas as pd
import numpy as np

## Carga de datos y exploración del respectivo dataframe

In [64]:
df_movie=pd.read_csv(r"Datasets/movies_dataset.csv",delimiter=",")

In [65]:
# Visualización de las doce primeras columnas
df_movie.iloc[:3,:12]

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   

     original_title                                           overview  \
0         Toy Story  Led by Woody, Andy's toys live happily in his ...   
1           Jumanji  When siblings Judy and Peter discover an encha...   
2  Grumpier Old Men  A family wedding reignites the ancient feud be...   

   popularity                       poster_path  
0   21.946943  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg  
1   17.015539  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  
2   11.712900  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg

In [66]:
# Visualización de las últimas doce columnas
df_movie.iloc[:3,12:]

production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   

                                production_countries release_date    revenue  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30  373554033   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15  262797249   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22          0   

   runtime                                   spoken_languages    status  \
0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   
1    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released   
2    101.0           [{'iso_639_1': 'en', 'name': 'English'}]  Released   

                                             tagline             title  video  \
0                                                NaN         Toy Story  False   
1          Roll the dice and unleash the excitement!           Jumanji  False   
2  Still Yelling. Still Fighting. Still Ready for...  Grumpier Old Men  False   

   vote_average  vote_count  
0           7.7        5415  
1           6.9        2413  
2           6.5          92

In [67]:
# Información del data frame
df_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45463 entries, 0 to 45462
Data columns (total 24 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45463 non-null  bool   
 1   belongs_to_collection  4491 non-null   object 
 2   budget                 45463 non-null  int64  
 3   genres                 45463 non-null  object 
 4   homepage               7779 non-null   object 
 5   id                     45463 non-null  int64  
 6   imdb_id                45446 non-null  object 
 7   original_language      45452 non-null  object 
 8   original_title         45463 non-null  object 
 9   overview               44509 non-null  object 
 10  popularity             45463 non-null  float64
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

* #### Observación: se procederá a eliminar las filas duplicadas en las cuales todos los valores de todas las columnas sean idénticos. Esto se realiza con el fin de garantizar la integridad y consistencia de los datos en el data frame. 

In [68]:
# Observando los valores duplicados
duplicates=df_movie[df_movie.duplicated(keep=False)].sort_values(by="id")
print(duplicates.shape[0])
duplicates.head(8)

33


adult belongs_to_collection  budget  \
7345   False                   NaN       0   
9165   False                   NaN       0   
24843  False                   NaN       0   
14012  False                   NaN       0   
22150  False                   NaN       0   
14000  False                   NaN       0   
8068   False                   NaN       0   
9327   False                   NaN       0   

                                                  genres  \
7345   [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...   
9165   [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...   
24843  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
14012  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
22150  [{'id': 28, 'name': 'Action'}, {'id': 27, 'nam...   
14000  [{'id': 28, 'name': 'Action'}, {'id': 27, 'nam...   
8068   [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   
9327   [{'id': 12, 'name': 'Adventure'}, {'id': 16, '...   

                                     homepage     id    imdb_id  \
7345                                      NaN   5511  tt0062229   
9165                                      NaN   5511  tt0062229   
24843            http://www.dealthemovie.com/  11115  tt0446676   
14012            http://www.dealthemovie.com/  11115  tt0446676   
22150  http://www.daysofdarknessthemovie.com/  18440  tt0499456   
14000  http://www.daysofdarknessthemovie.com/  18440  tt0499456   
8068                                      NaN  23305  tt0295682   
9327                                      NaN  23305  tt0295682   

      original_language    original_title  \
7345                 fr       Le Samouraï   
9165                 fr       Le Samouraï   
24843                en              Deal   
14012                en              Deal   
22150                en  Days of Darkness   
14000                en  Days of Darkness   
8068                 en       The Warrior   
9327                 en       The Warrior   

                                                overview  ...  release_date  \
7345   Hitman Jef Costello is a perfectionist who alw...  ...    1967-10-25   
9165   Hitman Jef Costello is a perfectionist who alw...  ...    1967-10-25   
24843  As an ex-gambler teaches a hot-shot college ki...  ...    2008-01-29   
14012  As an ex-gambler teaches a hot-shot college ki...  ...    2008-01-29   
22150  When a comet strikes Earth and kicks up a clou...  ...    2007-01-01   
14000  When a comet strikes Earth and kicks up a clou...  ...    2007-01-01   
8068   In feudal India, a warrior (Khan) who renounce...  ...    2001-09-23   
9327   In feudal India, a warrior (Khan) who renounce...  ...    2001-09-23   

      revenue runtime                           spoken_languages    status  \
7345    39481   105.0  [{'iso_639_1': 'fr', 'name': 'Français'}]  Released   
9165    39481   105.0  [{'iso_639_1': 'fr', 'name': 'Français'}]  Released   
24843       0    85.0   [{'iso_639_1': 'en', 'name': 'English'}]  Released   
14012       0    85.0   [{'iso_639_1': 'en', 'name': 'English'}]  Released   
22150       0    89.0   [{'iso_639_1': 'en', 'name': 'English'}]  Released   
14000       0    89.0   [{'iso_639_1': 'en', 'name': 'English'}]  Released   
8068        0    86.0    [{'iso_639_1': 'hi', 'name': 'हिन्दी'}]  Released   
9327        0    86.0    [{'iso_639_1': 'hi', 'name': 'हिन्दी'}]  Released   

                                                 tagline             title  \
7345   There is no solitude greater than that of the ...       Le Samouraï   
9165   There is no solitude greater than that of the ...       Le Samouraï   
24843                                                NaN              Deal   
14012                                                NaN              Deal   
22150                                                NaN  Days of Darkness   
14000                                                NaN  Days of Darkness   
8068                                                 NaN       The Warrior   
93

In [69]:
# Eliminando los duplicados
df_movie.drop_duplicates(inplace=True)
df_movie=df_movie.reset_index(drop=True)

## Transformaciones

#### 1) Desanidación de los campos anidados que se encuentran en forma de diccionarios

In [70]:
# Columnas en forma de diccionario
df_movie.iloc[:3,[1,3,12,13,17]]

belongs_to_collection  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...   
1                                                NaN   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   

                                production_countries  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   

                                    spoken_languages  
0           [{'iso_639_1': 'en', 'name': 'English'}]  
1  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  
2           [{'iso_639_1': 'en', 'name': 'English'}]

* Columna `belongs_to_collection`

Este campo almacena las rutas relativas de las imágenes de las películas. Sin embargo, estos datos no son necesarios, ya que no se requieren en ninguna etapa del proyecto. Además, algunas de las rutas ya no son funcionales y solo hay un número reducido de valores no nulos (4491). Por lo tanto, solo se considerarán los valores de las claves `name`.

In [76]:
df_movie.belongs_to_collection[0]

"{'id': 10194, 'name': 'Toy Story Collection', 'poster_path': '/7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg', 'backdrop_path': '/9FBwqcd9IRruEDUrTdcaafOMKUq.jpg'}"

In [78]:
# Aplicando la función 'name'
df_movie["belongs_to_collection"]=df_movie.belongs_to_collection.apply(lambda dic: eval(dic)["name"] if pd.notnull(dic) else dic)

* Columnas `genres`, `production_companies`, ``production_countries`` y `spoken_languages`

Se guardarán exclusivamente los valores de las claves ``name`` en una lista, ya que los ID no son necesarios. Al extraer únicamente los valores de la clave `name`, se simplifica y conserva la información más relevante y descriptiva de los campos correspondientes. Esta acción mejora la legibilidad, comprensión y utilidad de los datos en el contexto del análisis y la visualización de los datos.

In [79]:
print(df_movie.genres[0])
print(df_movie.production_companies[1])
print(df_movie.production_countries[6])
print(df_movie.spoken_languages[5])

[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]
[{'name': 'TriStar Pictures', 'id': 559}, {'name': 'Teitler Film', 'id': 2550}, {'name': 'Interscope Communications', 'id': 10201}]
[{'iso_3166_1': 'DE', 'name': 'Germany'}, {'iso_3166_1': 'US', 'name': 'United States of America'}]
[{'iso_639_1': 'en', 'name': 'English'}, {'iso_639_1': 'es', 'name': 'Español'}]


In [80]:
# Función que extrae los valores de las claves 'name'
def name(string):
    return [dic["name"] for dic in eval(string)]

In [81]:
# Aplicando la función 'name'
df_movie["genres"]=df_movie.genres.apply(name)
df_movie["production_companies"]=df_movie.production_companies.apply(name)
df_movie["production_countries"]=df_movie.production_countries.apply(name)
df_movie["spoken_languages"]=df_movie.spoken_languages.apply(name)

# Visualizando los resultados
df_movie.loc[:7,["belongs_to_collection","genres","production_companies","production_countries","spoken_languages"]]

belongs_to_collection                              genres  \
0            Toy Story Collection         [Animation, Comedy, Family]   
1                             NaN        [Adventure, Fantasy, Family]   
2       Grumpy Old Men Collection                   [Romance, Comedy]   
3                             NaN            [Comedy, Drama, Romance]   
4  Father of the Bride Collection                            [Comedy]   
5                             NaN    [Action, Crime, Drama, Thriller]   
6                             NaN                   [Comedy, Romance]   
7                             NaN  [Action, Adventure, Drama, Family]   

                                production_companies  \
0                          [Pixar Animation Studios]   
1  [TriStar Pictures, Teitler Film, Interscope Co...   
2                     [Warner Bros., Lancaster Gate]   
3           [Twentieth Century Fox Film Corporation]   
4       [Sandollar Productions, Touchstone Pictures]   
5  [Regency Enterprises, Forward Pass, Warner Bros.]   
6  [Paramount Pictures, Scott Rudin Productions, ...   
7                             [Walt Disney Pictures]   

                  production_countries     spoken_languages  
0           [United States of America]            [English]  
1           [United States of America]  [English, Français]  
2           [United States of America]            [English]  
3           [United States of America]            [English]  
4           [United States of America]            [English]  
5           [United States of America]   [English, Español]  
6  [Germany, United States of America]  [Français, English]  
7           [United States of America]   [English, Deutsch]

#### 2) Imputación de los valores nulos de los campos ``revenue`` y ``budget`` por el número 0

No existen valores nulos en estos campos

In [83]:
print(df_movie[df_movie.revenue.isnull()].shape[0])
print(df_movie[df_movie.budget.isnull()].shape[0])

0
0


#### 3) Conversión de las fechas al formato AAAA-mm-dd y creación de la columna ``release_year`` 

El único campo de fecha disponible se encuentra en la columna ``release_date``, el cual se convertirá al tipo 'date'. A partir de este campo, se extraerá el año de estreno de cada película.

In [84]:
df_movie["release_date"]=pd.to_datetime(df_movie.release_date)
df_movie["release_year"]=df_movie.release_date.dt.year

# Visualizando los resultados
df_movie.loc[:,["release_date","release_year"]]

release_date  release_year
0       1995-10-30        1995.0
1       1995-12-15        1995.0
2       1995-12-22        1995.0
3       1995-12-22        1995.0
4       1995-02-10        1995.0
...            ...           ...
45441          NaT           NaN
45442   2011-11-17        2011.0
45443   2003-08-01        2003.0
45444   1917-10-21        1917.0
45445   2017-06-09        2017.0

[45446 rows x 2 columns]

#### 4) Creación de la columna ``return``: ``revenue`` / ``budget``

* Observación: Si el campo ``budget`` es igual a cero, el valor de ``return`` será establecido como cero, ya que la división por cero no está definida y no es posible realizarla.

In [85]:
# La mayoría de películas tienen un presupuesto (budget) de cero
f"{(df_movie[df_movie.budget==0].budget.shape[0]/df_movie.shape[0])*100}%"

'80.44052281828984%'

In [86]:
# Creando la columna 'return'
df_movie["return"]=df_movie.apply(lambda x: round(x.revenue/x.budget,2) if x.budget!=0 else 0,axis=1)
df_movie["return"]

0        12.45
1         4.04
2         0.00
3         5.09
4         0.00
         ...  
45441     0.00
45442     0.00
45443     0.00
45444     0.00
45445     0.00
Name: return, Length: 45446, dtype: float64

#### 5) Eliminación de las columnas que no serán utilizadas: ``video``, ``imdb_id``, ``adult``, ``original_title``, `vote_count`,``poster_path`` y ``homepage``.

In [87]:
df_movie.drop(columns=["video","imdb_id","adult","original_title","vote_count","poster_path","homepage"],inplace=True)

* Visualizando el data frame resultante

In [88]:
df_movie

belongs_to_collection    budget                        genres  \
0                Toy Story Collection  30000000   [Animation, Comedy, Family]   
1                                 NaN  65000000  [Adventure, Fantasy, Family]   
2           Grumpy Old Men Collection         0             [Romance, Comedy]   
3                                 NaN  16000000      [Comedy, Drama, Romance]   
4      Father of the Bride Collection         0                      [Comedy]   
...                               ...       ...                           ...   
45441                             NaN         0               [Drama, Family]   
45442                             NaN         0                       [Drama]   
45443                             NaN         0     [Action, Drama, Thriller]   
45444                             NaN         0                            []   
45445                             NaN         0                            []   

           id original_language  \
0         862                en   
1        8844                en   
2       15602                en   
3       31357                en   
4       11862                en   
...       ...               ...   
45441  439050                fa   
45442  111109                tl   
45443   67758                en   
45444  227506                en   
45445  461257                en   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...   21.946943   
1      When siblings Judy and Peter discover an encha...   17.015539   
2      A family wedding reignites the ancient feud be...   11.712900   
3      Cheated on, mistreated and stepped on, the wom...    3.859495   
4      Just when George Banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45441        Rising and falling between a man and woman.    0.072051   
45442  An artist struggles to finish his work while a...    0.178241   
45443  When one of her hits goes wrong, a professiona...    0.903007   
45444  In a small town live two brothers, one a minis...    0.003503   
45445  50 years after decriminalisation of homosexual...    0.163015   

                                    production_companies  \
0                              [Pixar Animation Studios]   
1      [TriStar Pictures, Teitler Film, Interscope Co...   
2                         [Warner Bros., Lancaster Gate]   
3               [Twentieth Century Fox Film Corporation]   
4           [Sandollar Productions, Touchstone Pictures]   
...                                                  ...   
45441                                                 []   
45442                                      [Sine Olivia]   
45443                          [American World Pictures]   
45444                                        [Yermoliev]   
45445                                                 []   

             production_countries release_date    revenue  runtime  \
0      [United States of America]   1995-10-30  373554033     81.0   
1      [United States of America]   1995-12-15  262797249    104.0   
2      [United States of America]   1995-12-22          0    101.0   
3      [United States of America]   1995-12-22   81452156    127.0   
4      [United States of America]   1995-02-10   76578911    106.0   
...                           ...          ...        ...      ...   
45441                      [Iran]          NaT          0     90.0   
45442               [Philippines]   2011-11-17          0    360.0   
45443  [United States of America]   2003-08-01          0     90.0   
45444                    [Russia]   1917-10-21          0     87.0   
45445            [United Kingdom]   2017-06-09          0     75.0   

          spoken_languages    status  \
0                [English]  Released   
1      [English, Français]  Released   
2                [English]  Released   
3                [English]  Release

In [89]:
df_movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45446 entries, 0 to 45445
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   belongs_to_collection  4490 non-null   object        
 1   budget                 45446 non-null  int64         
 2   genres                 45446 non-null  object        
 3   id                     45446 non-null  int64         
 4   original_language      45435 non-null  object        
 5   overview               44492 non-null  object        
 6   popularity             45446 non-null  float64       
 7   production_companies   45446 non-null  object        
 8   production_countries   45446 non-null  object        
 9   release_date           45362 non-null  datetime64[ns]
 10  revenue                45446 non-null  int64         
 11  runtime                45189 non-null  float64       
 12  spoken_languages       45446 non-null  object        
 13  s

### Exportación del data frame "df_platform" en un archivo CSV para la realización de los endpoints

In [355]:
df_movie.to_csv("movie_transformation.csv",index=False)

---

In [40]:
import pandas as pd
import numpy as np

In [5]:
df=pd.read_csv(r"https://raw.githubusercontent.com/JersonGB22/ProyectoIndividual1_Henry/main/Datasets/movie_transformation.csv")
df

budget                              genres      id original_language  \
0      30000000   ['Animation', 'Comedy', 'Family']     862                en   
1      65000000  ['Adventure', 'Fantasy', 'Family']    8844                en   
2             0               ['Romance', 'Comedy']   15602                en   
3      16000000      ['Comedy', 'Drama', 'Romance']   31357                en   
4             0                          ['Comedy']   11862                en   
...         ...                                 ...     ...               ...   
45441         0                 ['Drama', 'Family']  439050                fa   
45442         0                           ['Drama']  111109                tl   
45443         0     ['Action', 'Drama', 'Thriller']   67758                en   
45444         0                                  []  227506                en   
45445         0                                  []  461257                en   

                                                overview  popularity  \
0      Led by Woody, Andy's toys live happily in his ...   21.946943   
1      When siblings Judy and Peter discover an encha...   17.015539   
2      A family wedding reignites the ancient feud be...   11.712900   
3      Cheated on, mistreated and stepped on, the wom...    3.859495   
4      Just when George Banks has recovered from his ...    8.387519   
...                                                  ...         ...   
45441        Rising and falling between a man and woman.    0.072051   
45442  An artist struggles to finish his work while a...    0.178241   
45443  When one of her hits goes wrong, a professiona...    0.903007   
45444  In a small town live two brothers, one a minis...    0.003503   
45445  50 years after decriminalisation of homosexual...    0.163015   

                                    production_companies  \
0                            ['Pixar Animation Studios']   
1      ['TriStar Pictures', 'Teitler Film', 'Intersco...   
2                     ['Warner Bros.', 'Lancaster Gate']   
3             ['Twentieth Century Fox Film Corporation']   
4       ['Sandollar Productions', 'Touchstone Pictures']   
...                                                  ...   
45441                                                 []   
45442                                    ['Sine Olivia']   
45443                        ['American World Pictures']   
45444                                      ['Yermoliev']   
45445                                                 []   

               production_countries release_date    revenue  runtime  \
0      ['United States of America']   1995-10-30  373554033     81.0   
1      ['United States of America']   1995-12-15  262797249    104.0   
2      ['United States of America']   1995-12-22          0    101.0   
3      ['United States of America']   1995-12-22   81452156    127.0   
4      ['United States of America']   1995-02-10   76578911    106.0   
...                             ...          ...        ...      ...   
45441                      ['Iran']          NaN          0     90.0   
45442               ['Philippines']   2011-11-17          0    360.0   
45443  ['United States of America']   2003-08-01          0     90.0   
45444                    ['Russia']   1917-10-21          0     87.0   
45445            ['United Kingdom']   2017-06-09          0     75.0   

              spoken_languages    status  \
0                  ['English']  Released   
1      ['English', 'Français']  Released   
2                  ['English']  Released   
3                  ['English']  Released   
4                  ['English']  Released   
...                        ...       ...   
45441                ['فارسی']  Released   
45442                     ['']  Released   
45443              ['English']  Released   
45444                       []  Released   
45445              ['English']  Released   

                                                 tagline

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45446 entries, 0 to 45445
Data columns (total 18 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   budget                45446 non-null  int64  
 1   genres                45446 non-null  object 
 2   id                    45446 non-null  int64  
 3   original_language     45435 non-null  object 
 4   overview              44492 non-null  object 
 5   popularity            45446 non-null  float64
 6   production_companies  45446 non-null  object 
 7   production_countries  45446 non-null  object 
 8   release_date          45362 non-null  object 
 9   revenue               45446 non-null  int64  
 10  runtime               45189 non-null  float64
 11  spoken_languages      45446 non-null  object 
 12  status                45365 non-null  object 
 13  tagline               20408 non-null  object 
 14  title                 45446 non-null  object 
 15  vote_average       

In [6]:
# Cantidad de películas producidas por un determinado país en determinado año
df.head(1)

budget                             genres   id original_language  \
0  30000000  ['Animation', 'Comedy', 'Family']  862                en   

                                            overview  popularity  \
0  Led by Woody, Andy's toys live happily in his ...   21.946943   

          production_companies          production_countries release_date  \
0  ['Pixar Animation Studios']  ['United States of America']   1995-10-30   

     revenue  runtime spoken_languages    status tagline      title  \
0  373554033     81.0      ['English']  Released     NaN  Toy Story   

   vote_average  release_year  return  
0           7.7        1995.0   12.45

In [ ]:
df[""]

In [34]:
country='United States of America'
year=2016

In [42]:
np.sort(df.release_year.dropna().unique())

(135,)

In [35]:
df[(df.production_countries.apply(lambda x: country in eval(x))) & (df.release_year==year)]

budget                                             genres      id  \
26545  100000000                   ['Adventure', 'Drama', 'Action']  271969   
26546  160000000                 ['Action', 'Adventure', 'Fantasy']   68735   
26553   58000000                  ['Action', 'Adventure', 'Comedy']  293660   
26556  250000000         ['Adventure', 'Action', 'Science Fiction']  271110   
26557  165000000  ['Action', 'Adventure', 'Fantasy', 'Science Fi...  284052   
...          ...                                                ...     ...   
45172          0          ['Action', 'Science Fiction', 'Thriller']  429524   
45194          0                                ['Comedy', 'Crime']  412103   
45236          0                                    ['Documentary']  390343   
45364          0                      ['Comedy', 'Horror', 'Drama']  434873   
45372          0                            ['TV Movie', 'Romance']  401687   

      original_language                                           overview  \
26545                en  A falsely accused nobleman survives years of s...   
26546                en  The peaceful realm of Azeroth stands on the br...   
26553                en  Deadpool tells the origin story of former Spec...   
26556                en  Following the events of Age of Ultron, the col...   
26557                en  After his career is destroyed, a brilliant but...   
...                 ...                                                ...   
45172                en  When drones arrive in a flash, slaughtering hu...   
45194                en  A faithful, word-for-word recreation of one co...   
45236                en  A probing portrait of Chris Burden, an artist ...   
45364                en  In the throes of a zombie apocalypse, A troubl...   
45372                en  Hopeless romantic Matt (Brandon W. Jones) deci...   

       popularity                               production_companies  \
26545   11.510210  ['Paramount Pictures', 'Metro-Goldwyn-Mayer (M...   
26546   12.572718  ['Universal Pictures', 'Atlas Entertainment', ...   
26553  187.860492  ['Twentieth Century Fox Film Corporation', 'Ma...   
26556  145.882135  ['Studio Babelsberg', 'Marvel Studios', 'Walt ...   
26557   43.847654         ['Walt Disney Pictures', 'Marvel Studios']   
...           ...                                                ...   
45172    1.451207                                  ['Cinetel Films']   
45194    0.760622  ['Williams Street', 'Harmonius Claptrap', 'Sta...   
45236    0.638724                              ['Magnolia Pictures']   
45364    8.471866  ['Digital Interference Productions', 'Grasswoo...   
45372    0.253497                     ['Larry Levinson Productions']   

                                    production_countries release_date  \
26545                       ['United States of America']   2016-08-17   
26546  ['Canada', 'China', 'Japan', 'United States of...   2016-05-25   
26553                       ['United States of America']   2016-02-09   
26556                       ['United States of America']   2016-04-27   
26557                       ['United States of America']   2016-10-25   
...                                                  ...          ...   
45172                       ['United States of America']   2016-03-30   
45194                       ['United States of America']   2016-07-24   
45236                       ['United States of America']   2016-04-16   
45364                       ['United States of America']   2016-10-08   
45372                       ['United States of America']   2016-06-04   

          revenue  runtime                             spoken_languages  \
26545    94061311    125.0                                  ['English']   
26546   433677183    123.0                                  ['English']   
26553   783112979    108.0                                  ['English']   
26556  1153304495    147.0  ['Română', 'English', 'Deutsch', 'Pусский']   
26557  